prime-router/src/test/kotlin/ReportTests.kt

```diff
@@ -132,4 +130,140 @@ class ReportTests {
+        val schema = Schema(
+            name = "test",
+            topic = "test",
+            elements = listOf(
+                Element("last_name"), Element("first_name")
+            )
+        )
+        val report = Report(
+            schema = schema,
+            values = listOf(listOf("smith", "sarah"), listOf("jones", "mary"), listOf("white", "roberta"),),
+            source = TestSource
+        )
+        val strategies = mapOf(
+            "last_name" to Report.SynthesizeStrategy.PASSTHROUGH,
+            "first_name" to Report.SynthesizeStrategy.PASSTHROUGH
+        )
+        // act
+        val synthesizedReport = report.synthesizeData(strategies)
+        // assert
+        assertEquals(3, synthesizedReport.itemCount)
+        assertEquals("smith", synthesizedReport.getString(0, "last_name"))
+        assertEquals("jones", synthesizedReport.getString(1, "last_name"))
+        assertEquals("white", synthesizedReport.getString(2, "last_name"))
+        assertEquals("sarah", synthesizedReport.getString(0, "first_name"))
+        assertEquals("mary", synthesizedReport.getString(1, "first_name"))
+        assertEquals("roberta", synthesizedReport.getString(2, "first_name"))
+    }
+
+    @Test
+    fun `test synthesize data with blank strategy`() {
+        // arrange
+        val schema = Schema(
+            name = "test",
+            topic = "test",
+            elements = listOf(
+                Element("last_name"), Element("first_name"), Element("ssn")
+            )
+        )
+        val report = Report(
+            schema = schema,
+            values = listOf(
+                listOf("smith", "sarah", "000000000"),
+                listOf("jones", "mary", "000000000"),
+                listOf("white", "roberta", "000000000"),
+            ),
+            source = TestSource
+        )
+        val strategies = mapOf(
+            "last_name" to Report.SynthesizeStrategy.PASSTHROUGH,
+            "first_name" to Report.SynthesizeStrategy.PASSTHROUGH,
+            "ssn" to Report.SynthesizeStrategy.BLANK,
+        )
+        // act
+        val synthesizedReport = report.synthesizeData(strategies)
+        // assert
+        assertEquals(3, synthesizedReport.itemCount)
+        assertEquals("smith", synthesizedReport.getString(0, "last_name"))
+        assertEquals("jones", synthesizedReport.getString(1, "last_name"))
+        assertEquals("white", synthesizedReport.getString(2, "last_name"))
+        assertEquals("sarah", synthesizedReport.getString(0, "first_name"))
+        assertEquals("mary", synthesizedReport.getString(1, "first_name"))
+        assertEquals("roberta", synthesizedReport.getString(2, "first_name"))
+        assertEquals("", synthesizedReport.getString(0, "ssn"))
+        assertEquals("", synthesizedReport.getString(1, "ssn"))
+        assertEquals("", synthesizedReport.getString(2, "ssn"))
+    }
+
+    // ignoring this test for now because shuffling is non-deterministic
+    @Test
```

Comment:
You could just pull both the orig and the shuffled data into Sets, then the Sets should be equal - all the original names are still there, with the same number of rows.  That seems like a pretty good test.

Reply:
Could definitely do that.